src/communication/network_interface_sr.c

```diff
@@ -10223,3 +10226,209 @@ ssession_stop_attached_threads (void *session)
+  char *ptr;
+  LOG_LSA start_lsa;
+  time_t input_time;
+  uint64_t b_start_lsa;
+  int error;
+
+  ptr = or_unpack_int64 (request, &input_time);
+
+  error = cdc_find_lsa (thread_p, &input_time, &start_lsa);
+
+  ptr = or_pack_int (reply, error);
+  ptr = or_pack_log_lsa (ptr, &start_lsa);
+  or_pack_int64 (ptr, input_time);
+
+  (void) css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
+
+  return;
+}
+
+void
+scdc_initialize (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  OR_ALIGNED_BUF (OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  char *ptr;
+  int error;
+  int max_log_item, extraction_timeout, all_in_cond, num_extraction_user, num_extraction_class;
+  uint64_t *extraction_classoids = NULL;
+  char **extraction_user = NULL;
+  int extraction_user_len = 0;
+
+  char *dummy_user = NULL;
+  int timeout;
+
+  ptr = or_unpack_int (request, &max_log_item);
+  ptr = or_unpack_int (ptr, &extraction_timeout);
+  ptr = or_unpack_int (ptr, &all_in_cond);
+  ptr = or_unpack_int (ptr, &num_extraction_user);
+
+  if (num_extraction_user > 0)
+    {
+      extraction_user = (char **) malloc (sizeof (char *) * num_extraction_user);
+      if (extraction_user == NULL)
+	{
+	  error = ER_OUT_OF_VIRTUAL_MEMORY;
+	  goto end;
+	}
+
+      for (int i = 0; i < num_extraction_user; i++)
+	{
+	  ptr = or_unpack_string (ptr, &dummy_user);
+	  extraction_user_len = strlen (dummy_user);
+	  extraction_user[i] = (char *) malloc (extraction_user_len + 1);
+
+	  if (extraction_user[i] == NULL)
+	    {
+	      for (int j = 0; j <= i; j++)
+		{
+		  free (extraction_user[j]);
+		}
+	      free (extraction_user);
+	      db_private_free_and_init (NULL, dummy_user);
+
+	      error = ER_OUT_OF_VIRTUAL_MEMORY;
+	      goto end;
+	    }
+
+	  memcpy (extraction_user[i], dummy_user, extraction_user_len);
+	  db_private_free_and_init (NULL, dummy_user);
+	  extraction_user[i][extraction_user_len] = '\0';
+	}
+    }
+
+  ptr = or_unpack_int (ptr, &num_extraction_class);
+
+  if (num_extraction_class > 0)
+    {
+      extraction_classoids = (uint64_t *) malloc (sizeof (uint64_t) * num_extraction_class);
+      if (extraction_classoids == NULL)
+	{
+	  if (num_extraction_user > 0)
+	    {
+	      for (int i = 0; i < num_extraction_user; i++)
+		{
+		  free (extraction_user[i]);
+		}
+
+	      free (extraction_user);
+	    }
+
+	  error = ER_OUT_OF_VIRTUAL_MEMORY;
+	  goto end;
+	}
+      for (int i = 0; i < num_extraction_class; i++)
+	{
+	  ptr = or_unpack_int64 (ptr, (INT64 *) & extraction_classoids[i]);
+	}
+    }
+
+  error =
+    cdc_set_configuration (max_log_item, timeout, all_in_cond, extraction_user, num_extraction_user,
+			   extraction_classoids, num_extraction_class);
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+  error = cdc_initialize ();
+  if (error != NO_ERROR)
+    {
+      goto end;
+    }
+
+end:
+
+  ptr = or_pack_int (reply, error);
```

Comment:
How about using the error_code iso error? I think it is consistent compare to the other function in this file.